### PR TITLE
Fix validator ownership proportion problem

### PR DESCRIPTION
**Pull request by nebolax**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/tests/unit/test_eth2.py [hunk 1]
```diff
@@ -940,3 +940,25 @@ def test_validator_daily_stats_with_genesis_event(
...
+    dbeth2 = DBEth2(database)
+    with database.user_write() as cursor:
+        dbeth2.add_validators(cursor, [Eth2Validator(
+            index=9,
+            public_key=Eth2PubKey('0xb016e31f633a21fbe42a015152399361184f1e2c0803d89823c224994af74a561c4ad8cfc94b18781d589d03e952cd5b'),  # noqa: E501
+            ownership_proportion=FVal(0.5),
+        )])
+    result = eth2.fetch_eth1_validator_data(
+        addresses=[ADDR1],
+    )
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_eth2.py:963]: It fits in one line

```suggestion
    result = eth2.fetch_eth1_validator_data(addresses=[ADDR1])
```

**Review by yabirgb — COMMENTED** [review]
A few comments and questions

File: rotkehlchen/chain/ethereum/modules/eth2/eth2.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def fetch_eth1_validator_data(
             self,
             addresses: list[ChecksumEvmAddress],
     ) -> list[ValidatorID]:
-        """Query all eth1 addresses for their validators and get all corresponding deposits.
+        """Query all eth1 addresses for their validators.
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/chain/ethereum/modules/eth2/eth2.py:139]: Yes. And deposits were not used here for some time, it was just a remnant

File: docs/changelog.rst [hunk 3]
```diff
@@ -19,6 +19,7 @@ Changelog
 * :feature:`-` Swaps made via uniswap v3 auto routers (both v1 and v2) will now be decoded correctly.
 * :feature:`3231` Optimism is now supported. Optimism balances will be shown and optimism transactions will be decoded.
 * :feature:`1756` Uniswap and sushiswap pool join/exit events are now properly decoded and taken into account during PnL report.
+* :bug:`-` Now customized ownership proportions of validators owned by eth1 addresses will be respected.
```
Comment (LefterisJP) [inline docs/changelog.rst:22]: I have corrected this multiple times. Do not start a sentence with `Now`.

```suggestion
* :bug:`-` Customized ownership proportions of validators owned by eth1 addresses will now be properly respected.
```

File: rotkehlchen/chain/ethereum/modules/eth2/eth2.py [hunk 4]
```diff
@@ -147,39 +147,35 @@ def fetch_eth1_validator_data(
...
 
-            pubkeys.update([x.public_key for x in validators])
-            all_validators.extend(validators)
-            # if we already have any of those validators in the DB, no need to query deposits
-            with self.database.conn.read_ctx() as cursor:
-                tracked_validators = dbeth2.get_validators(cursor)
-            tracked_pubkeys = [x.public_key for x in tracked_validators]
-            new_validators = [
+            new_validator_ids = [x for x in validators if x.public_key not in tracked_pubkeys]
+            new_validators.extend([
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/modules/eth2/eth2.py:168]: ouch

File: rotkehlchen/chain/ethereum/modules/eth2/eth2.py (hunk 2 above)
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/eth2/eth2.py:139]: Also perhaps rename function to show you are writing to the DB. `fetch_and_update_eth1_validator_data`

```suggestion
        """Query all eth1 addresses for their validators and any newly detected validators are added to the DB
```

File: rotkehlchen/tests/unit/test_eth2.py [hunk 5]
```diff
@@ -940,3 +940,45 @@ def test_validator_daily_stats_with_genesis_event(
...
+                index=9,
+                public_key=Eth2PubKey('0xb016e31f633a21fbe42a015152399361184f1e2c0803d89823c224994af74a561c4ad8cfc94b18781d589d03e952cd5b'),  # noqa: E501
+                ownership_proportion=FVal(0.5),
+            ),
+            Eth2Validator(
+                index=1647,
+                public_key=Eth2PubKey('0xb80777b022a115579f22674883996d0a904e51afaf0ddef4e577c7bc72ec4e14fc7714b8c58fb77ceb7b5162809d1475'),  # noqa: E501
+                ownership_proportion=FVal(0.7),
+            ),
+            Eth2Validator(  # This validator is not owned by neither of the addresses
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_eth2.py:974]:
```suggestion
            Eth2Validator(  # This validator is tracked but not owned by any of the addresses
```

File: rotkehlchen/tests/unit/test_eth2.py [hunk 6]
```diff
@@ -940,3 +940,45 @@ def test_validator_daily_stats_with_genesis_event(
...
+                index=1757,
+                public_key=Eth2PubKey('0xac3d4d453d58c6e6fd5186d8f231eb00ff5a753da3669c208157419055c7c562b7e317654d8c67783c656a956927209d'),  # noqa: E501
+                ownership_proportion=FVal(0.9),
+            ),
+        ])
+    result = eth2.fetch_eth1_validator_data(addresses=[ADDR1, ADDR2])
+    assert result[0].ownership_proportion == FVal(0.5), 'Proportion from the DB should be used'
+    assert result[1].ownership_proportion == FVal(0.7), 'Proportion from the DB should be used'
+    assert result[2].ownership_proportion == FVal(0.9), 'Proportion from the DB should be used'
+    assert result[3].ownership_proportion == ONE, 'Since this validator is new, the proportion should be ONE'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_eth2.py:984]: Is this validator id 997? Perhaps you can also assert on the ids? Also no network calls here, right? Is all mocked?

Comment (nebolax) [reply rotkehlchen/tests/unit/test_eth2.py:984]: Will add ids checks :+1:

Everything is mocked, yes.

File: docs/changelog.rst (hunk 3 above)
Comment (nebolax) [reply docs/changelog.rst:22]: Ah(((

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**